commit 735d265d796dfd1bc25a1c7fa06cfc3907bedbc6
Author: Trent Mick <trentm@gmail.com>
Date:   2019-06-14T10:18:18-07:00 (4 months ago)
    
    TRITON-1744 'sdcadm experimental remove-ca' crash: "Cannot read property 'Symbol(Symbol.iterator)' of undefined"
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 3654916..32011b8 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,10 @@
 
 # sdcadm Changelog
 
+## 1.30.1
+
+- TRITON-1744 Fix a crash in `sdcadm experimental remove-ca` if a given CNAPI
+  server record does not have an "agents" field.
 - TRITON-1748 Ensure 'sdcadm experimental remove-ca' does trace logging to file.
 
 ## 1.30.0
diff --git a/lib/procedures/remove-services.js b/lib/procedures/remove-services.js
index ece91c4..49123fb 100644
--- a/lib/procedures/remove-services.js
+++ b/lib/procedures/remove-services.js
@@ -250,14 +250,16 @@ RemoveServicesProcedure.prototype.prepare = function prepare(opts, cb) {
                 let agentsToRemove = [];
                 let cnAgentInfo;
 
-                for (let agent of server.agents) {
-                    if (agent.name === 'cn-agent') {
-                        cnAgentInfo = agent;
-                    }
-                    let sfsn = self.serversFromSvcName[agent.name];
-                    if (sfsn) {
-                        sfsn.push(server);
-                        agentsToRemove.push(agent.name);
+                if (server.agents) {
+                    for (let agent of server.agents) {
+                        if (agent.name === 'cn-agent') {
+                            cnAgentInfo = agent;
+                        }
+                        let sfsn = self.serversFromSvcName[agent.name];
+                        if (sfsn) {
+                            sfsn.push(server);
+                            agentsToRemove.push(agent.name);
+                        }
                     }
                 }
 
diff --git a/package.json b/package.json
index de0d92d..4248235 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a Triton Data Center",
-  "version": "1.30.0",
+  "version": "1.30.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
